libraries/AP_HAL_ChibiOS/EventSource.cpp

```diff
@@ -13,7 +14,7 @@ bool EventSource::wait(uint64_t duration, AP_HAL::EventHandle *evt_handle)
     if (duration == 0) {
         ret = chEvtWaitAnyTimeout(evt_mask, TIME_IMMEDIATE);
     } else {
-        ret = chEvtWaitAnyTimeout(evt_mask, chTimeUS2I(duration));
+        ret = chEvtWaitAnyTimeout(evt_mask, chTimeUS2I(MAX(CH_CFG_ST_TIMEDELTA,duration)));
```

Comment:
Is CH_CFG_ST_TIMEDELTA in ticks or us?

Reply:
its ticks, but on all but a few peripheral boards on F1 ticks==microseconds